### PR TITLE
feat: Conference Scheduling quickstart

### PR DESCRIPTION
**Pull request by zepfred**
## Description of the change

This PR adds a new quickstart `conference-scheduling`.

Screenshot from 2024-03-18 15-05-33
Screenshot from 2024-03-18 15-05-36
Screenshot from 2024-03-18 15-05-38
Screenshot from 2024-03-18 15-05-40
Screenshot from 2024-03-18 15-05-43
Screenshot from 2024-03-18 15-05-45
Screenshot from 2024-03-18 15-05-55
Screenshot from 2024-03-18 15-06-55
Screenshot from 2024-03-18 15-06-59
Screenshot from 2024-03-18 15-07-02
Screenshot from 2024-03-18 15-07-04
Screenshot from 2024-03-18 15-07-07
Screenshot from 2024-03-18 15-07-09
Screenshot from 2024-03-18 15-07-13

This solves TimefoldAi/timefold-solver#83

### Development

- [x] The changes have been covered with tests, if necessary.
- [x] You have a green build, with the exception of the flaky tests.
- [x] UI and JS files are fully tested, the user interface works for all modules affected by your changes (e.g., solve and analyze buttons).
- [x] The network calls work for all modules affected by your changes (e.g., solving a problem).
- [x] The console messages are validated for all modules affected by your changes.

### Code Review

- [x] This pull request includes an explanatory title and description.
- [x] The GitHub issue is linked.
- [ ] At least one other engineer has approved the changes.
- [ ] After PR is merged, inform the reporter.

### CODE REVIEW

File: use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/domain/Talk.java [hunk 1]
```diff
@@ -0,0 +1,611 @@
...
+        int leftSize = left.size();
+        if (leftSize == 0) {
+            return 0;
+        }
+        int rightSize = right.size();
+        if (rightSize == 0) {
+            return 0;
+        }
+        Set<T> smaller = leftSize < rightSize ? left : right;
+        Set<T> other = smaller == left ? right : left;
```
Comment (triceo) [inline use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/domain/Talk.java]: This is an optimization which, although useful, probably doesn't need to be here.
This is a quickstart - let's not overcomplicate it.
(The original example was suffering from it a bit.)

File: use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/ConferenceSchedulingResource.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        description = "Conference Scheduling service assigning rooms and timeslots for conference talks.")
+@Path("schedules")
+public class ConferenceSchedulingResource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConferenceSchedulingResource.class);
+
+    private final SolverManager<ConferenceSchedule, String> solverManager;
+    private final SolutionManager<ConferenceSchedule, HardSoftScore> solutionManager;
+
+    // TODO: Without any "time to live", the map may eventually grow out of memory.
```
Comment (triceo) [inline use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/ConferenceSchedulingResource.java]: Do we intend to do anything about this TODO?

File: use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/DemoDataGenerator.java [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    private static final String RECORDED_TAG = "Recorded";
+    private static final String LARGE_TAG = "Large";
+    // Theme tags
+    private static final List<String> THEME_TAGS = List.of("Theme1", "Theme2", "Theme3");
+    // Sector tags
+    private static final List<String> SECTOR_TAGS = List.of("Sector1", "Sector2", "Sector3");
+    // Audience tags
+    private static final List<String> AUDIENCE_TAGS = List.of("Audience1", "Audience2", "Audience3");
+    // Content tags
+    private static final List<String> CONTENT_TAGS = List.of("Content1", "Content2", "Content3", "Content4", "Content5");
```
Comment (triceo) [inline use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/DemoDataGenerator.java]: Can we get some nice names for these tags? Themes could be "Optimization", "AI", ... Sectors could be colors. ("Green sector", "Blue sector", ...) Etc.

File: use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/DemoDataGenerator.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    private final Random random = new Random();
```
Comment (triceo) [inline use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/DemoDataGenerator.java]: Let's make this predictable. Give it a fixed seed.

File: use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/solver/ConferenceSchedulingConstraintProvider.java [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+/**
+ * Provides the constraints for the conference scheduling problem.
+ * <p>
+ * Makes heavy use of CS expand() functionality to cache computation results,
+ * except in cases where doing so less is efficient than recomputing the result.
+ * That is the case in filtering joiners.
+ * In this case, it is better to reduce the size of the joins even at the expense of duplicating some calculations.
+ * In other words, time saved by caching those calculations is far outweighed by the time spent in unrestricted joins.
+ */
+public class ConferenceSchedulingConstraintProvider implements ConstraintProvider {
```
Comment (triceo) [inline use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/solver/ConferenceSchedulingConstraintProvider.java:82]: Although I understand that doing a different justification for each of these constraints would have been crazy, maybe the current granularity is lower than it ought to be? 

At least `TalkTagJustification` would IMO be a good addition.

**Review by triceo — COMMENTED** [review]
Initial comments.

File: use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/ConferenceSchedulingResource.java (hunk 2 above)
Comment (zepfred) [reply use-cases/conference-scheduling/src/main/java/org/acme/conferencescheduling/rest/ConferenceSchedulingResource.java]: I have added a function to remove the oldest jobs from the map.

**Review by triceo — APPROVED** [review]

**Pull request merged**